### PR TITLE
Address many suggestions in the 'Features' projects.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
Will be easier to review with whitespace off.  Done as multiple independent commits to make reviewing easier.

By and large i was applying relevant suggestions in these projects.  For a small number of suggestions i did the bulk fix (i.e. "remove unused usings"), for almost all the rest i performed them manually one at a time to ensure it was actually an improvement in all locations.  i.e. there were some times when i felt a suggestion detracted from readability.  In those cases i didn't apply the fix.

**Comment by CyrusNajmabadi** [comment]
Tagging @jasonmalinowski @dotnet/roslyn-ide

File: src/EditorFeatures/Core/Implementation/TextStructureNavigation/AbstractTextStructureNavigatorProvider.TextStructureNavigator.cs [hunk 1]
```diff
@@ -69,7 +69,7 @@ private TextExtent GetExtentOfWordWorker(SnapshotPoint position, CancellationTok
                 // If at the end of the file, go back one character so stuff works
                 if (position == textLength && position > 0)
                 {
-                    position = position - 1;
+                    position -= 1;
```
Comment (jcouv) [inline src/EditorFeatures/Core/Implementation/TextStructureNavigation/AbstractTextStructureNavigatorProvider.TextStructureNavigator.cs]: >position -= 1; [](start = 20, length = 14)

`position--;` ?

File: src/EditorFeatures/Core/Implementation/TodoComment/TodoItem.cs [hunk 2]
```diff
@@ -55,8 +55,7 @@ internal sealed class TodoItem
 
         public override bool Equals(object obj)
         {
-            TodoItem other = obj as TodoItem;
-            if (other == null)
+            if (!(obj is TodoItem other))
```
Comment (jcouv) [inline src/EditorFeatures/Core/Implementation/TodoComment/TodoItem.cs]: >if (!(obj is TodoItem other)) [](start = 12, length = 29)

consider inverting `if`

File: src/Features/Core/Portable/Completion/Providers/AbstractDocCommentCompletionProvider.cs [hunk 3]
```diff
@@ -211,14 +210,12 @@ protected IEnumerable<CompletionItem> GetTopLevelItems(ISymbol symbol, TSyntax s
                 items.AddRange(GetParameterItems(symbol.GetParameters(), syntax, ParameterElementName));
                 items.AddRange(GetParameterItems(symbol.GetTypeParameters(), syntax, TypeParameterElementName));
 
-                var property = symbol as IPropertySymbol;
-                if (property != null && !existingTopLevelTags.Contains(ValueElementName))
+                if (symbol is IPropertySymbol property && !existingTopLevelTags.Contains(ValueElementName))
```
Comment (jcouv) [inline src/Features/Core/Portable/Completion/Providers/AbstractDocCommentCompletionProvider.cs]: >property [](start = 46, length = 8)

is this variable used?

File: src/Features/CSharp/Portable/CodeRefactorings/SyncNamespace/CSharpChangeNamespaceService.cs [hunk 4]
```diff
@@ -358,8 +358,7 @@ protected override async Task<SyntaxNode> TryGetApplicableContainerFromSpanAsync
             }
 
             return container;
-
-            bool ContainsNamespaceDeclaration(SyntaxNode node)
+            static bool ContainsNamespaceDeclaration(SyntaxNode node)
```
Comment (jcouv) [inline src/Features/CSharp/Portable/CodeRefactorings/SyncNamespace/CSharpChangeNamespaceService.cs]: >ContainsNamespaceDeclaration [](start = 24, length = 28)

It's probably good to keep the empty line above to separate the local function.

File: src/Features/CSharp/Portable/InvertIf/CSharpInvertIfCodeRefactoringProvider.cs [hunk 5]
```diff
@@ -242,8 +242,7 @@ protected override bool IsSingleStatementStatementRange(StatementRange statement
             }
 
             return isSingleStatement(statementRange.FirstStatement);
-
-            bool isSingleStatement(StatementSyntax statement)
+            static bool isSingleStatement(StatementSyntax statement)
```
Comment (jcouv) [inline src/Features/CSharp/Portable/InvertIf/CSharpInvertIfCodeRefactoringProvider.cs]: >isSingleStatement [](start = 24, length = 17)

Empty line seemed beneficial. 
Also, I think IDE naming convention is PascalCase for local functions. It hurts me to say that ;-)

File: src/Features/CSharp/Portable/MakeMethodAsynchronous/CSharpMakeMethodAsynchronousCodeFixProvider.cs [hunk 6]
```diff
@@ -113,8 +112,7 @@ protected override bool IsAsyncSupportingFunctionSyntax(SyntaxNode node)
             }
 
             return newReturnType.WithTriviaFrom(returnTypeSyntax);
-
-            TypeSyntax MakeGenericType(string type, ITypeSymbol typeArgumentFrom)
+            static TypeSyntax MakeGenericType(string type, ITypeSymbol typeArgumentFrom)
```
Comment (jcouv) [inline src/Features/CSharp/Portable/MakeMethodAsynchronous/CSharpMakeMethodAsynchronousCodeFixProvider.cs]: >MakeGenericType [](start = 30, length = 15)

consider restoring empty line. Same in `IsYield` below

File: src/Features/CSharp/Portable/UseLocalFunction/CSharpUseLocalFunctionCodeFixProvider.cs [hunk 7]
```diff
@@ -226,14 +226,13 @@ public override Task RegisterCodeFixesAsync(CodeFixContext context)
...
-            int i = 0;
+            var i = 0;
 
             return parameterList != null
                 ? parameterList.ReplaceNodes(parameterList.Parameters, (parameterNode, _) => PromoteParameter(parameterNode, delegateMethod.Parameters.ElementAtOrDefault(i++)))
                 : SyntaxFactory.ParameterList(SyntaxFactory.SeparatedList(delegateMethod.Parameters.Select(parameter =>
                     PromoteParameter(SyntaxFactory.Parameter(parameter.Name.ToIdentifierToken()), parameter))));
-
-            ParameterSyntax PromoteParameter(ParameterSyntax parameterNode, IParameterSymbol delegateParameter)
+            static ParameterSyntax PromoteParameter(ParameterSyntax parameterNode, IParameterSymbol delegateParameter)
```
Comment (jcouv) [inline src/Features/CSharp/Portable/UseLocalFunction/CSharpUseLocalFunctionCodeFixProvider.cs]: >PromoteParameter [](start = 35, length = 16)

same here

File: src/EditorFeatures/Core/Implementation/TextStructureNavigation/AbstractTextStructureNavigatorProvider.TextStructureNavigator.cs (hunk 1 above)
Comment (CyrusNajmabadi) [reply src/EditorFeatures/Core/Implementation/TextStructureNavigation/AbstractTextStructureNavigatorProvider.TextStructureNavigator.cs]: Yes.  it would def be good for the feature to detect this pattern and suggest -- and ++ accordingly.  But i'll treat that as more of a feature suggestion :)

File: src/Features/CSharp/Portable/CodeRefactorings/SyncNamespace/CSharpChangeNamespaceService.cs (hunk 4 above)
Comment (CyrusNajmabadi) [reply src/Features/CSharp/Portable/CodeRefactorings/SyncNamespace/CSharpChangeNamespaceService.cs]: hrrmm... this def seems to be a defect in the fixer.  It should not be removing that blank line.

File: src/Features/CSharp/Portable/InvertIf/CSharpInvertIfCodeRefactoringProvider.cs (hunk 5 above)
Comment (CyrusNajmabadi) [reply src/Features/CSharp/Portable/InvertIf/CSharpInvertIfCodeRefactoringProvider.cs]: done.

File: src/Features/Core/Portable/Completion/Providers/AbstractDocCommentCompletionProvider.cs (hunk 3 above)
Comment (CyrusNajmabadi) [reply src/Features/Core/Portable/Completion/Providers/AbstractDocCommentCompletionProvider.cs]: nope.  addressed.  our fixer could do better here in these cases.

File: src/Features/CSharp/Portable/CodeRefactorings/SyncNamespace/CSharpChangeNamespaceService.cs (hunk 4 above)
Comment (miloush) [reply src/Features/CSharp/Portable/CodeRefactorings/SyncNamespace/CSharpChangeNamespaceService.cs]: Is this tracked somewhere? I believe I experience fixes removing empty lines or final line breaks quite often.

**Comment by sharwell** [comment]
This is likely to conflict heavily with https://github.com/dotnet/roslyn/pull/34477.

File: src/VisualStudio/CSharp/Impl/Microsoft.VisualStudio.LanguageServices.CSharp.csproj [hunk 8]
```diff
@@ -60,14 +60,44 @@
       <DesignTime>True</DesignTime>
       <DependentUpon>CSharpVSResources.resx</DependentUpon>
     </Compile>
+    <Compile Update="Options\AdvancedOptionPage.cs">
+      <SubType>Code</SubType>
+    </Compile>
```
Comment (sharwell) [inline src/VisualStudio/CSharp/Impl/Microsoft.VisualStudio.LanguageServices.CSharp.csproj]: 📝 These should not be added in this commit

File: src/Features/CSharp/Portable/AddBraces/CSharpAddBracesDiagnosticAnalyzer.cs [hunk 9]
```diff
@@ -230,7 +230,7 @@ private static bool IsConsideredMultiLine(SyntaxNode statement, SyntaxNode embed
         }
 
         /// <summary>
-        /// Determines whether <paramref name="embeddedStatement"/> should use braces under a
+        /// Determines whether <paramref name="statement"/> should use braces under a
```
Comment (sharwell) [inline src/Features/CSharp/Portable/AddBraces/CSharpAddBracesDiagnosticAnalyzer.cs]: 📝 This one snuck in

File: src/Features/Core/Portable/ExtractMethod/AbstractSyntaxTriviaService.cs [hunk 10]
```diff
@@ -95,12 +95,14 @@ public ITriviaSavedResult SaveTriviaAroundSelection(SyntaxNode root, TextSpan te
...
+            var triviaList = new Dictionary<TriviaLocation, IEnumerable<SyntaxTrivia>>
+            {
+                [TriviaLocation.BeforeBeginningOfSpan] = triviaAtBeginning.Item1,
+                [TriviaLocation.AfterBeginningOfSpan] = triviaAtBeginning.Item2,
 
-            triviaList[TriviaLocation.BeforeEndOfSpan] = triviaAtEnd.Item1;
-            triviaList[TriviaLocation.AfterEndOfSpan] = triviaAtEnd.Item2;
+                [TriviaLocation.BeforeEndOfSpan] = triviaAtEnd.Item1,
+                [TriviaLocation.AfterEndOfSpan] = triviaAtEnd.Item2
+            };
```
Comment (sharwell) [inline src/Features/Core/Portable/ExtractMethod/AbstractSyntaxTriviaService.cs]: 📝 Change is unrelated to current commit

File: src/Features/Core/Portable/CodeFixes/Suppression/AbstractSuppressionCodeFixProvider.PragmaBatchFixHelpers.cs [hunk 11]
```diff
@@ -129,9 +129,9 @@ private static class PragmaBatchFixHelpers
 
             private static void UpdateDiagnosticSpans(ImmutableArray<Diagnostic> diagnostics, Dictionary<Diagnostic, TextSpan> currentDiagnosticSpans, IEnumerable<TextChange> textChanges)
             {
-                static bool isPriorSpan(TextSpan span, TextChange textChange) => span.End <= textChange.Span.Start;
-                static bool isFollowingSpan(TextSpan span, TextChange textChange) => span.Start >= textChange.Span.End;
-                static bool isEnclosingSpan(TextSpan span, TextChange textChange) => span.Contains(textChange.Span);
+                static bool IsPriorSpan(TextSpan span, TextChange textChange) => span.End <= textChange.Span.Start;
+                static bool IsFollowingSpan(TextSpan span, TextChange textChange) => span.Start >= textChange.Span.End;
+                static bool IsEnclosingSpan(TextSpan span, TextChange textChange) => span.Contains(textChange.Span);
```
Comment (sharwell) [inline src/Features/Core/Portable/CodeFixes/Suppression/AbstractSuppressionCodeFixProvider.PragmaBatchFixHelpers.cs]: 😕 Are these not used?

File: src/Features/Core/Portable/CodeFixes/Suppression/AbstractSuppressionCodeFixProvider.PragmaBatchFixHelpers.cs [hunk 12]
```diff
@@ -154,14 +153,14 @@ private static void UpdateDiagnosticSpans(ImmutableArray<Diagnostic> diagnostics
...
-                            if (isFollowingSpan(originalSpan, textChange))
+                            if (IsFollowingSpan(originalSpan, textChange))
                             {
                                 // Following span.
                                 var newStart = currentSpan.Start + delta;
                                 currentSpan = new TextSpan(newStart, currentSpan.Length);
                                 currentDiagnosticSpans[diagnostic] = currentSpan;
                             }
-                            else if (isEnclosingSpan(originalSpan, textChange))
+                            else if (IsEnclosingSpan(originalSpan, textChange))
```
Comment (sharwell) [inline src/Features/Core/Portable/CodeFixes/Suppression/AbstractSuppressionCodeFixProvider.PragmaBatchFixHelpers.cs]: 📝 Some changes snuck into the wrong commit here

File: src/Compilers/Core/AnalyzerDriver/AnalyzerExceptionDescriptionBuilder.cs [hunk 13]
```diff
@@ -14,8 +14,7 @@ internal static class AnalyzerExceptionDescriptionBuilder
 
         public static string CreateDiagnosticDescription(this Exception exception)
         {
-            var aggregateException = exception as AggregateException;
-            if (aggregateException != null)
+            if (exception is AggregateException aggregateException)
```
Comment (sharwell) [inline src/Compilers/Core/AnalyzerDriver/AnalyzerExceptionDescriptionBuilder.cs]: 📝 A few here seemed to sneak into the wrong commit

**Comment by CyrusNajmabadi** [comment]
> This is likely to conflict heavily with #34477.

I wouldn't think there would be conflicts here.  the changes should be the same on both our machines.

File: src/VisualStudio/CSharp/Impl/Microsoft.VisualStudio.LanguageServices.CSharp.csproj (hunk 8 above)
Comment (CyrusNajmabadi) [reply src/VisualStudio/CSharp/Impl/Microsoft.VisualStudio.LanguageServices.CSharp.csproj]: i keep reverting these, and VS keeps changing them back...  will manually revert outside of VS.  but i'm not sure what's up here.

**Comment by CyrusNajmabadi** [comment]
Ok. Reverted the files.

**Comment by CyrusNajmabadi** [comment]
@sharwell Can we move forward with this?

**Comment by sharwell** [comment]
@CyrusNajmabadi I'd like to start with the smaller changes in #34477. However, code cleanup changes like this (non-critical work) are blocked until we can restore the integration test leg to a properly functioning state. Some bugs crept in that are causing consistent failures there.

**Comment by CyrusNajmabadi** [comment]
@sharwell are the integration tests working again?  Can we move forward on this?

**Comment by CyrusNajmabadi** [comment]
I recommend hiding whitespace changes on this.  That helps with several of hte cahnges.

**Comment by CyrusNajmabadi** [comment]
@sharwell @tmat @jcouv is tehre a way to proceed with this?

File: src/EditorFeatures/Test/RenameTracking/RenameTrackingTestState.cs [hunk 14]
```diff
@@ -112,14 +112,12 @@ internal sealed class RenameTrackingTestState : IDisposable
             if (languageName == LanguageNames.CSharp)
             {
                 _codeFixProvider = new CSharpRenameTrackingCodeFixProvider(
-                    Workspace.ExportProvider.GetExport<Host.IWaitIndicator>().Value,
```
Comment (jcouv) [inline src/EditorFeatures/Test/RenameTracking/RenameTrackingTestState.cs]: I didn't understand this change (not familiar enough with area or logic) so I can't comment.

File: src/VisualStudio/CSharp/Impl/Microsoft.VisualStudio.LanguageServices.CSharp.csproj (hunk 8 above)
Comment (jcouv) [reply src/VisualStudio/CSharp/Impl/Microsoft.VisualStudio.LanguageServices.CSharp.csproj]: Still needs to be reverted.

**Comment by sharwell** [comment]
Aside from the previous issue related to another open PR, the following are my main concerns:

1. Some of the fixes were automatically applied, some were manually applied, and some were automatically applied followed by manual corrections (e.g. to formatting). It's not possible to determine which change is which, and therefore work towards improving the automatic fixes to cover more scenarios.
1. I would expect at most two commits per subject matter: one for purely automatic changes and a second, if necessary, for manual changes not covered by automatic.
1. Some mistakes were made in the separation of fixes into individual commits.
1. The changes are attributed to @CyrusNajmabadi and not to @dotnet-bot. Typically all purely automatic refactorings would be assigned to @dotnet-bot, and manual refactorings would be applied to the author. This is visible in #34477, for example.

**Comment by CyrusNajmabadi** [comment]
> Some mistakes were made in the separation of fixes into individual commits.

Note: all i used here was the built-in VS git tooling.  I'm not sure how it works.  But i've definitely committed, only to find that post-commit there are some files remaining.  Not sure what causes this.

**Comment by CyrusNajmabadi** [comment]
> I don't believe these issues can be corrected without a substantial rewrite of this pull request and possibly breaking it up into multiple pull requests

This seems excessive.  I don't see how the problems listed are that much of an issue.  For example, i've seen in many PRs the same sorts of things happening, without the team having an issue, or having problems merging along with actual *product changes*.

This is simply trying to extract this out so we have less cases of normal-PRs that include suggestion-fixes.  It seems stranger that we would be more restrictive about this than with PRs where we have to look more closely to see what actual logic changes are happening.

--

Note: i am happy to break this up if that will help.  But my primary concern here is simply around speed and efficacy of the review process.  It can routinely take weeks or *months* to get attention on reviews.  So breaking things up simply means that what could be a fast and simple approach to merging, becomes literal *years* as we try to get each PR in, going through a multi-week back-and-forth.  

With the IDE's current turnaround on this sort of thing, i think ti would be far better to just get things in, even if stuff like "it's attributed to Cyrus, and not dotnet-bot" are there.

**Comment by CyrusNajmabadi** [comment]
Tagging @jinujoseph for thoughts here.

**Comment by CyrusNajmabadi** [comment]
> Normally pull requests that just change code style are closed unless they are internally coordinated, but I'll leave the final decision to one of the project managers.

Note, i don't think it's appropriate to describe this PR as one that "just change[s] code style".  This PR brings these IDE projects *in line* with the actual code-style preferences specified by the project's .editorconfig settings.  I'm not randomly going and enforcing some "cyrus belief" about what things should look like.  All i'm doing is actually just accepting what the tooling says it should be, based on what the project has indicated it would prefer.

Part of the reason i'm doing this is because the current state of affairs is poor.  It's harder to use the product because there are notifications everywhere i go that the code is non-conformant.  I can't really use the task-list to find issues because it's got tens-of-thousands of messages about non-conformancy *everywhere*.  And as i'm going through files, i'm constantly running into messages from teh IDE about what should be fixed up.

I'm going this not for "code styles sake", but so that there is less distraction and so that when i actually write code, i can easily find the code-style breaches in what i've just written without being overloaded with all the rest of the messages about the rest of the doc/project/solution.

@jinujoseph for what may be possible to do here to help things out.

**Comment by CyrusNajmabadi** [comment]
@mavasani Would def like eyes on this.  Tried to break it into many commits taht i then submitted to make things simpler.  Thanks!

**Comment by mavasani** [comment]
> Part of the reason i'm doing this is because the current state of affairs is poor. It's harder to use the product because there are notifications everywhere i go that the code is non-conformant. I can't really use the task-list to find issues because it's got tens-of-thousands of messages about non-conformancy everywhere. And as i'm going through files, i'm constantly running into messages from teh IDE about what should be fixed up.

I believe this is a strong reason to take this PR - it makes it easier to dogfood IDE analyzers within Roslyn.sln itself

**Comment by CyrusNajmabadi** [comment]
@mavasani can you merge in when you have a chance?

**Comment by mavasani** [comment]
@CyrusNajmabadi The PR needs to be targeting release\dev16.3 branch. Master branch is in M2 mode.

**Comment by CyrusNajmabadi** [comment]
@mavasani @jinujoseph when does master open up again for normal work?  Is there a reason that master is repurposed in this fashion?

**Comment by CyrusNajmabadi** [comment]
@mavasani Which of these should i be using:

image

**Comment by mavasani** [comment]
`release/dev16.3-preview1`

**Comment by CyrusNajmabadi** [comment]
@mavasani i tried rebasing to that and it pulled in an additional 5k changes.  Any suggestions on how to target that branch?  Also, it would be *really* nice if you could let hte community know about this.  Not having 'master' be the development branch is *super* weird and difficult to deal with.  I have many PRs made out aainst master that i've been keeping up to date for months.  It's unclear how to manage them across having to move all of them (and their downstream deps) over to another branch.

File: eng/Version.Details.xml [hunk 15]
```diff
@@ -3,9 +3,9 @@
   <ProductDependencies>
   </ProductDependencies>
   <ToolsetDependencies>
-    <Dependency Name="Microsoft.DotNet.Arcade.Sdk" Version="1.0.0-beta.19272.13">
+    <Dependency Name="Microsoft.DotNet.Arcade.Sdk" Version="1.0.0-beta.19323.4">
       <Uri>https://github.com/dotnet/arcade</Uri>
-      <Sha>86e674361bdcefecbd8199ab62d0b1a6cb25703d</Sha>
+      <Sha>9946534da4f73e6242ca105f6798ab58119c9ab0</Sha>
```
Comment (CyrusNajmabadi) [inline eng/Version.Details.xml]: @mavasani I did not change this.

File: eng/common/PSScriptAnalyzerSettings.psd1 [hunk 16]
```diff
@@ -0,0 +1,11 @@
...
+    IncludeRules=@('PSAvoidUsingCmdletAliases',
+                   'PSAvoidUsingWMICmdlet',
+                   'PSAvoidUsingPositionalParameters',
+                   'PSAvoidUsingInvokeExpression',
+                   'PSUseDeclaredVarsMoreThanAssignments',
+                   'PSUseCmdletCorrectly',
+                   'PSStandardDSCFunctionsInResource',
+                   'PSUseIdenticalMandatoryParametersForDSC',
+                   'PSUseIdenticalParametersForDSC')
+}
```
Comment (CyrusNajmabadi) [inline eng/common/PSScriptAnalyzerSettings.psd1]: @mavasani I did not change this. 

I'm not sure what's going on.  It looks like this preview branch has diverged a *lot* from master.

File: eng/common/PublishToPackageFeed.proj [hunk 17]
```diff
@@ -54,6 +54,7 @@
       <TargetStaticFeed Condition="'$(ArtifactsCategory.ToUpper())' == 'WINDOWSDESKTOP'">https://dotnetfeed.blob.core.windows.net/dotnet-windowsdesktop/index.json</TargetStaticFeed>
       <TargetStaticFeed Condition="'$(ArtifactsCategory.ToUpper())' == 'NUGETCLIENT'">https://dotnetfeed.blob.core.windows.net/nuget-nugetclient/index.json</TargetStaticFeed>
       <TargetStaticFeed Condition="'$(ArtifactsCategory.ToUpper())' == 'ASPNETENTITYFRAMEWORK6'">https://dotnetfeed.blob.core.windows.net/aspnet-entityframework6/index.json</TargetStaticFeed>
+      <TargetStaticFeed Condition="'$(ArtifactsCategory.ToUpper())' == 'ASPNETBLAZOR'">https://dotnetfeed.blob.core.windows.net/aspnet-blazor/index.json</TargetStaticFeed>
```
Comment (CyrusNajmabadi) [inline eng/common/PublishToPackageFeed.proj]: thoughts/suggestions on what to do?  i would prefer to not have to revert all these individual changes.

**Comment by mavasani** [comment]
> i tried rebasing to that and it pulled in an additional 5k changes. Any suggestions on how to target that branch? Also, it would be really nice if you could let hte community know about this. Not having 'master' be the development branch is super weird and difficult to deal with. I have many PRs made out aainst master that i've been keeping up to date for months. It's unclear how to manage them across having to move all of them (and their downstream deps) over to another branch.

Tagging @vatsalyaagrawal

**Comment by CyrusNajmabadi** [comment]
@mavasani @jinujoseph i don't know what to do with this PR now. :-/

**Comment by jinujoseph** [comment]
master should be available for 16.3 work early next week. 
@vatsalyaagrawal @agocke  for awareness on the issue.

**Comment by CyrusNajmabadi** [comment]
> master should be available for 16.3 work early next week.

In the future, can we keep master always open for all work?  It seems artificially limiting to put restrictions on it (and seems to violate the concept of what a 'master' branch should be).  

If restrictions are needed, those should be pulled out into separate branches.  Then it can be clear what hte restrictions of those particular branches are.  i.e. you go into this branch if you want to make it into 'x-release', and you need to meet the 'x-release' bar there.  

Having 'master' change just means there's no way (especially for someone external) for anyone to know what actual purpose 'master' serves.  Hence, it is not the 'master' branch anymore.

**Comment by CyrusNajmabadi** [comment]
@mavasani can you help shephard this into the right branch?

**Comment by CyrusNajmabadi** [comment]
@mavasani @jasonmalinowski can you merge this in?

**Comment by mavasani** [comment]
@CyrusNajmabadi can you please resolve the merge conflicts?

**Comment by CyrusNajmabadi** [comment]
Ugh.  This is very frustrating.  The PR was created almost two months ago, and was maintained against master.  When no one is willing to look at or merge in PRs, it just adds so much work and pain for contributers.  Having that much time to simply apply simple automated fixes to the codebase just sucks and makes the idea of contributing even slightly interesting PRs very unpalatable.  Tagging @jinujoseph for visibility on this as it does not feel like the team considers the cost and pain to contributers with the current approach on community PRs at all.

**Comment by CyrusNajmabadi** [comment]
Note: i cannot actually build roslyn due to:https://github.com/dotnet/roslyn/issues/36840

**Comment by mavasani** [comment]
@CyrusNajmabadi seeing the following errors `[error]src\Features\Core\Portable\IntroduceVariable\AbstractIntroduceVariableService.State.cs(188,1): error CS8300: Merge conflict marker encountered`

**Comment by CyrusNajmabadi** [comment]
@mavasani On it.

**Comment by mavasani** [comment]
@CyrusNajmabadi Thank you for working through different hurdles on this PR, and sorry for the additional work for you from branch changes.

> Ugh. This is very frustrating. The PR was created almost two months ago, and was maintained against master. When no one is willing to look at or merge in PRs, it just adds so much work and pain for contributers. Having that much time to simply apply simple automated fixes to the codebase just sucks and makes the idea of contributing even slightly interesting PRs very unpalatable. Tagging @jinujoseph for visibility on this as it does not feel like the team considers the cost and pain to contributers with the current approach on community PRs at all.

@vatsalyaagrawal @jinujoseph I completely agree with Cyrus that master should be a permanently open branch targeting next big feature update. However, if the team feels otherwise, then next time we switch master to a release-specific branch, I would suggest we do the following when we create a new branch out of master which is for future feature updates (like release\dev16.3-preview1 created recently):
1. Update all open community PRs to target the newly created branch instead of master and add a comment on the PR about the same
2. Broadcast the message on the Roslyn channels, such as gitter, so all community members are aware about the change. This should be for both events where master becomes a release-specific branch and again when master becomes an open branch for feature work
3. It would even be good if any new PR creation in GitHub can also target this future branch by default, instead of master. I am not sure GitHub supports such a configuration on new PRs.

**Comment by CyrusNajmabadi** [comment]
Thanks!
